apis/management.cattle.io/v3/multi_cluster_app_types.go

```diff
@@ -13,3 +16,67 @@ type GlobalDNS struct {
 	TTLSeconds   int64  `json:"ttl" norman:"default=300"`
 	ProviderName string `json:"providerName,omitempty" norman:"required"`
 }
+
+type MultiClusterApp struct {
+	types.Namespaced
+
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiClusterAppSpec   `json:"spec,omitempty"`
+	Status MultiClusterAppStatus `json:"status,omitempty"`
+}
+
+type MultiClusterAppSpec struct {
+	HelmChartReference HelmChartReference `json:"helmChartReference,omitempty"`
+	TemplateReference  TemplateReference  `json:"templateReference,omitempty"`
+	Targets            []Target           `json:"targets,omitempty" norman:"required"`
+}
+
+type MultiClusterAppStatus struct {
+	HealthState string `json:"healthState,omitempty" norman:"required"`
+}
+
+type HelmChartReference struct {
+	ChartRepositoryURL string `json:"chartRepositoryUrl,omitempty"`
+	ChartReference     string `json:"chartReference,omitempty"`
+	ChartVersion       string `json:"chartVersion,omitempty"`
+	ReleaseName        string `json:"releaseName,omitempty"`
+	ReleaseNamespace   string `json:"releaseNamespace,omitempty"`
+}
+
+type TemplateReference struct {
+	TemplateName    string `json:"templateName,omitempty"`
+	TemplateVersion string `json:"templateVersion,omitempty"`
+}
+
+type Target struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TargetSpec   `json:"spec,omitempty" norman:"required"`
+	Status TargetStatus `json:"status,omitempty" norman:"required"`
+}
+
+type TargetSpec struct {
+	ClusterName   string            `json:"clusterName,omitempty"`
+	ClusterConfig ClusterConfig     `json:"clusterConfig,omitempty"`
+	Answers       map[string]string `json:"answers,omitempty"`
```

Comment:
Lets add answers on a global level as well. So user can define a super set on a global level, and override some of it on per cluster basis